client/my-sites/plugins/index.js

```diff
@@ -57,43 +70,65 @@ export default function() {
 		} );
 
 		if ( config.isEnabled( 'manage/plugins/upload' ) ) {
-			page( '/plugins/upload', sites );
-			page( '/plugins/upload/:site_id', siteSelection, navigation, pluginsController.upload );
+			page( '/plugins/upload', sites, makeLayout, clientRender );
+			page(
+				'/plugins/upload/:site_id',
+				siteSelection,
+				makeNavigation,
+				pluginsController.upload,
+				makeLayout,
+				clientRender
+			);
 		}
 
-		page( '/plugins', siteSelection, navigation, pluginsController.browsePlugins );
+		page(
+			'/plugins',
+			siteSelection,
+			makeNavigation,
+			pluginsController.browsePlugins,
+			makeLayout,
+			clientRender
+		);
 
 		page(
 			'/plugins/manage/:site?',
 			siteSelection,
-			navigation,
+			makeNavigation,
 			ifSimpleSiteThenRedirectTo( '/plugins' ),
 			pluginsController.plugins,
-			sites
+			// sites,
```

Comment:
Something in `sites` middleware is messing up with plugins section and I don't get these routes to render without commenting this out (routes just get stuck at site-selector).

Since route `http://calypso.localhost:3000/plugins/:site` attempts to load _a plugin called :site_ (instead of a plugin catalog), there might be something funny going on with routes as well.

I need to have a closer look at:
- `sites`: on first scan it didn't look suspicious.
- Route definitions and use of `next()` in case it's getting stuck anywhere.

Reply:
I noticed similar issues with my-sites/domains so concentrating now to properly test that site selector.

_Edit: moved comment to right place :-)_

Reply:
> Something in `sites` middleware is messing up with plugins section and I don't get these routes to render without commenting this out (routes just get stuck at site-selector).

Maybe that `ifSimpleSiteThenRedirectTo` thing? (Wild guess)

Anyway, the `sites` controller is supposed to render a site selection screen for Calypso sections and pages that don't work in multi-site mode (= most of them). Now if you try these routes in production, e.g. https://wordpress.com/plugins/manage, that seems to be a perfectly fine multi-site page: The user is shown all plugins used on all of their JP pages -- no sites selector is shown. So it looks like at least for `/plugins/manage`, the `sites` middleware is a spurious leftover that shouldn't be there.

Reply:
To expand a bit on nomenclature: multi-site sections/pages are those that make sense with data accumulated from all of a user's sites -- most notably posts, pages, themes, and plugins -- cf. the items in the sidebar at https://wordpress.com/stats/day (Be sure you actually have more than only one site when checking this out.)

Compare that to single-site mode, https://wordpress.com/stats/day/example.wordpress.com -- you'll see way more sidebar entries. Now if you click on a non-multi-site-enabled sidebar entry (= Calypso section), e.g. Sharing, and then use the site switcher on top of the sidebar to switch to 'All Sites' -- you'll be prompted with another site selector taking up the entire screen. _This_ is what the `sites` controller does. It only makes sense for non-multi-site-enabled routes.

Reply:
Thanks for elaborating on this feature, it made it easier to understand and debug.

I ended up refactoring a bit more because several things were on the way for me See PR https://github.com/Automattic/wp-calypso/pull/20312